pkg/gcv/validator.go

```diff
@@ -220,8 +222,60 @@ func (v *Validator) handleReview(ctx context.Context, idx int, asset *validator.
 	}
 }
 
+func ancestryPath(ancestors []string) string {
+	cnt := len(ancestors)
+	revAncestors := make([]string, len(ancestors))
+	for idx := 0; idx < cnt; idx++ {
+		revAncestors[cnt-idx-1] = ancestors[idx]
+	}
+	return strings.Join(revAncestors, "/")
+}
+
+const (
+	ancestryPathKey = "ancestry_path"
+	ancestorsKey    = "ancestors"
+)
+
+func (v *Validator) fixAncestry(input map[string]interface{}) error {
+	if _, found := input[ancestryPathKey]; found {
+		return nil
+	}
+
+	ancestorsIface, found := input[ancestorsKey]
+	if !found {
+		return errors.Errorf("no ancestors field: %s", input)
+	}
+	ancestors, ok := ancestorsIface.([]interface{})
+	if !ok {
+		return errors.Errorf("ancestors field not array type: %s", input)
+	}
+	cnt := len(ancestors)
+	revAncestors := make([]string, len(ancestors))
```

Comment:
Why are we reversing this twice (both in `ancestryPath` func and here)?

Reply:
Whoops, that wasn't supposed to happen.  Nice catch.